### DISCUSSION TITLE
Bicycle Routing crashed my phone

### DISCUSSION THREAD

**Discussion by BugsBunnySan** [osmandapp/OsmAnd#17913] [Maps Related]
I was trying to use OSMand (version 4.5.10, pro subscription) on my.phone (Samsung A71, newest software version) to create a bycicle route from Bad Waldsee through Riedlingen to Reutlingen (all of these towns are in Germany/Baden-Württemberg about 120km appart). I have installed/downloaded the app last week (around 6.August 2023) and downloaded the maps then and some additional maps today.

I've previously ridden my bike between these towns, so, there is definitivly a route. I just recently started using the OSMand app, so  wanted to see what it would suggest vs. how I rode it.

Here's what happened:

OSMand took forever to try to find a route. It got to about 95%, starting using 100% of all my phone's CPU (or RAM?, anyhow, my phone stopped responding to any input). 
OSMand did show a message in the routing screen that was still open, claiming it couldn't find a route. If the app.had just stopped there, that would've been really disappointing already (it's really not a difficult route to find) but the app was apparently still trying to fnd a route, since it was still using all my phone's resources aand the little blue progress indicator bar waa still there...

This then crashed the System UI app, which closed and then restarted (I couldn't stop OSMand because the phone was still unreaponsive).
Shortly after, my phone even did a hard reboot...

Thia seems really bad to me, to the point of making the app unusable for me, if everytime I try to use it to plot a route it might just crash my phone...

I am trying to like this app, since it has some nice features, but I can't really use it like this...

Cheers,
Sebastian Haas

**Comment by mikehgentry** [comment]
To be honest for long distance routing I tend to just use BRouter - it's not only faster but produces better routing for bikes in my opinion.

https://osmand.net/docs/user/navigation/routing/brouter/

Regarding your crash, it'd perhaps be helpful to try to do a kind of binary bisect: cut the route in half, and see if either half still causes the crash. If either half does, cut that half in half again... Repeat until you've found the area causing it. (edit: of course this may not help if it's not obvious which way it's going to route you... but it's worth a try)

If it's only the whole route causing the crash, you could perhaps try increasing the memory available to the router, which is a switch in the development plugin.

**Reply by BugsBunnySan** [reply]
Oh wow, using BRouter really is blazingly fast (and, as you said, produced a route I'm really happy with).
Cheers for suggesting that :)

I did increase that memory limit as well, just in case.

So, I do have a workaround now that's even better then before. I still think some bug fixing needs to happen in OSMand though, if its routing does run out of menory, it should really stop and not crash the phone.... well, IMHO...

**Reply by mikehgentry** [reply]
Indeed, it should be fixed, but it's hard to fix things until the specifics are narrowed down. 

Normally OsmAnd offers to send a crash report after you restart it, otherwise there's an option to do so in the 'Help' menu.

So if you can reliably reproduce the crash, that'd probably be useful. Even more so if you can establish that it works if you've allocated lots of memory, and doesn't if you haven't, or find a specific area that routing through causes crashing.

**Reply by DmitryAlexei** [reply]
@mikehgentry good suggestion about Brouter! Online routing engines can be also used (https://osmand.net/docs/user/navigation/routing/online-routing).

When using offline routing engine for long distances (more than 50-70 km for bicycle) it is also suggested to use intermediate points https://osmand.net/docs/user/troubleshooting/navigation#how-to-calculate-routes-longer-than-250km

**Comment by BugsBunnySan** [comment]
I thonk it didn't offer to send that crash report because it made my whole phone shutdown and reboot... 

Also, IMHO, though good advice with the splitting of routes (which I did do, actually) and using BRouter and such, the fact that OSMAnd's routing crashes without a chance of recovery is not so nice... 

Is there a chance to recover the logfiles even in such a case where my phone gets rebooted? I might give it another try to see if I can obtain a logfile to attach here or something...

Cheers,
Sebastian Haas

**Reply by DmitryAlexei** [reply]
Please, see https://docs.osmand.net/docs/user/troubleshooting/crash-logs#send-tombstone-files-android-only

**Comment by BugsBunnySan** [comment]
I created a bug report like it said and looked for and found the tombstone files inside it, howrver, they were all weeks (or more) older than the crash... so, I don't think I have files.from the crash of OSMAnd....

23 Aug 2023 08:42:46 DmitryAlexei ***@***.***>:
